Jenkinsfile

```diff
@@ -1104,7 +1082,13 @@ pipeline {
                             sconsBuild(parallel_build: true,
                                        scons_args: 'PREFIX=/opt/daos TARGET_TYPE=release BUILD_TYPE=debug',
                                        build_deps: 'no'))
-                        unstash('nltr')
+                        script {
+                            try {
+                                unstash('nltr')
+                            } catch (e) {
+                                print 'Unstash failed, ignoring'
```

Comment:
I realize this is still a draft PR, but as you get closer to something you'd want to land, it might be useful to expand this message into something that will be meaningful for the average developer.  Maybe indicate the consequences of the stash not being present (i.e. [how] coverage reporting will be affected) and what they can do to remedy this (i.e. don't skip stage [name] if you want to avoid this, etc.)

As for the `Jenkinsfile` size issue and the general use of `script {}` blocks (yes, I know they are used elsewhere), it would probably be better to write a `pipeline-lib` function that allows `unstash` to fail with a provided error message.  That would reduce this 7 line script block down into a single `pipeline-lib` function call.